packages/flutter_tools/lib/src/commands/build.dart

```diff
@@ -43,6 +45,11 @@ abstract class BuildSubCommand extends FlutterCommand {
   @override
   @mustCallSuper
   Future<FlutterCommandResult> runCommand() async {
+    final String channel = FlutterVersion.instance.getBranchName();
+    if (channel == 'master') {
+      printStatus('🐉', newline: false, color: TerminalColor.red);
+      printStatus(' This is the $channel channel. Shipping apps from this channel is not recommended as it has not been as heavily tested as the stable channel. To build using the stable channel, consider using:\n    "flutter channel stable".');
```

Comment:
trivial nit: since you have the colon, you don't need the trailing period.

nit: line break this string

Reply:
(also if you have it on its own line you don't need the quote marks)